src/ParametricOptInterface.jl

```diff
@@ -1255,6 +1420,7 @@ function MOI.set(
     empty_objective_function_caches!(model)
     if !function_has_parameters(model, f)
         MOI.set(model.optimizer, attr, f)
+        model.original_objective_function = f
```

Comment:
We could save it before the if and then simply delete it from the middle of the function